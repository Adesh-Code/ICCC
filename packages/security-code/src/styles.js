const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
