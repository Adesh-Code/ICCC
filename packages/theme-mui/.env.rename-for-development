# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE