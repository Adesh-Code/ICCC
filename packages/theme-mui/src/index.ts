// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
