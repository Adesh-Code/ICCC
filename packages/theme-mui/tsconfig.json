{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
