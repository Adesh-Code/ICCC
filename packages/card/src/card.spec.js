import React from "react";
import CardActionArea from "@mui/material/CardActionArea";
import CardActions from "@mui/material/CardActions";
import CardContent from "@mui/material/CardContent";
import CardMedia from "@mui/material/CardMedia";
import Card, { CardDivider } from "./card";

import { render } from "@testing-library/react";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = (options) => {
  const args = {
    children: "Card body",
    ...options
  };

  return render(
    <ThemeProvider theme={light}>
      <Card {...args} />
    </ThemeProvider>
  );
};

const testMediaUrl =
  "https://swc.autodesk.com/pharmacopeia/img/placeholders/960x640.png";

beforeEach(() => {
  // eslint-disable-next-line no-console
  console.error = jest.fn((error) => {
    throw new Error(error);
  });
});

describe("card component", () => {
  it("requires children for card", () => {
    expect(() =>
      arrange({
        children: null
      })
    ).toThrow();
  });

  it("renders a card", () => {
    const { container } = arrange();
    expect(container).toMatchSnapshot();
  });

  it("renders a quiet card", () => {
    const { container } = arrange({
      quiet: true
    });
    expect(container.firstChild).toHaveClass("MuiCard--quiet");
  });

  it("renders a framed card", () => {
    const { container } = arrange({
      framed: true
    });
    expect(container.firstChild).toHaveClass("MuiCard--framed");
  });

  it("renders a quiet and framed card", () => {
    const { container } = arrange({
      framed: true,
      quiet: true
    });
    expect(container.firstChild).toHaveClass("MuiCard--quiet");
    expect(container.firstChild).toHaveClass("MuiCard--framed");
  });

  it("renders an outlined card", () => {
    expect(() =>
      arrange({
        quiet: false
      })
    ).toBeTruthy();
  });

  it("renders a card with custom class name", () => {
    const { container } = arrange({
      className: "test-classname"
    });
    expect(container.firstChild).toHaveClass("test-classname");
  });

  it("renders a card with two column class added", () => {
    const { container } = arrange({
      twoColumns: true
    });
    expect(container.firstChild).toHaveClass("MuiCard--twoColumns");
  });
});

describe("card content component", () => {
  it("renders a card content", () => {
    const { container } = render(<CardContent>Hello</CardContent>);
    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });
});

describe("card actions component", () => {
  it("renders a card actions", () => {
    const { container } = render(<CardActions>Action</CardActions>);
    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });
});

describe("card action area component", () => {
  it("renders a card action area", () => {
    const { container } = render(
      <CardActionArea href="https://www.autodesk.com">Action</CardActionArea>
    );
    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });
});

describe("card media component", () => {
  it("renders a card media", () => {
    const { container } = render(<CardMedia image={testMediaUrl} />);
    expect(container).toMatchSnapshot();
    expect(container).toBeTruthy();
  });
});

describe("card divider component", () => {
  it("renders", () => {
    const { container } = render(<CardDivider />);
    expect(container).toMatchSnapshot();
  });
});
