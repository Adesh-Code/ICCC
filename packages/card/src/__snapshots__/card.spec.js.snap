// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card action area component renders a card action area 1`] = `
<div>
  <a
    class="MuiButtonBase-root MuiCardActionArea-root emotion-0"
    href="https://www.autodesk.com"
    tabindex="0"
  >
    Action
    <span
      class="MuiCardActionArea-focusHighlight emotion-1"
    />
    <span
      class="MuiTouchRipple-root emotion-2"
    />
  </a>
</div>
`;

exports[`card actions component renders a card actions 1`] = `
<div>
  <div
    class="MuiCardActions-root MuiCardActions-spacing emotion-0"
  >
    Action
  </div>
</div>
`;

exports[`card component renders a card 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiCard-root  DhigCard--appearance--default emotion-0"
  >
    Card body
  </div>
</div>
`;

exports[`card content component renders a card content 1`] = `
<div>
  <div
    class="MuiCardContent-root emotion-0"
  >
    Hello
  </div>
</div>
`;

exports[`card divider component renders 1`] = `
<div>
  <div
    class="DhigCard--divider"
  >
    <hr
      class="MuiDivider-root MuiDivider-fullWidth MuiDivider-light emotion-0"
    />
  </div>
</div>
`;

exports[`card media component renders a card media 1`] = `
<div>
  <div
    class="MuiCardMedia-root emotion-0"
    role="img"
    style="background-image: url(https://swc.autodesk.com/pharmacopeia/img/placeholders/960x640.png);"
  />
</div>
`;
